editor/components/post-publish-panel/toggle.js

```diff
@@ -40,6 +40,21 @@ function PostPublishPanelToggle( {
 		return <PostPublishButton forceIsDirty={ forceIsDirty } forceIsSaving={ forceIsSaving } />;
 	}
 
+	if ( ! hasPublishAction ) {
+		return (
+			<Button
+				className="editor-post-publish-panel__toggle"
+				isPrimary
+				onClick={ onToggle }
+				aria-expanded={ isOpen }
+				disabled={ ! isButtonEnabled }
+				isBusy={ isSaving && isPublished }
+			>
+				{ isBeingScheduled ? __( 'Schedule…' ) : __( 'Submit for Review…' ) }
```

Comment:
Same here, it looks like the only difference is the label, so the check should be done inside the button.

Also, I remember that it was an intended change to show "Publish" even for contributors. The idea is to keep the button small and wen you open the panel, you'll see the right message. cc @jasmussen